activerecord/test/cases/encryption/configurable_test.rb

```diff
@@ -65,6 +67,14 @@ class ActiveRecord::Encryption::ConfigurableTest < ActiveRecord::EncryptionTestC
     assert_includes application.config.filter_parameters, "catchphrase"
   end
 
+  test "installing autofiltered parameters will work with child classes" do
+    application = OpenStruct.new(config: OpenStruct.new(filter_parameters: []))
+    ActiveRecord::Encryption.install_auto_filtered_parameters_hook(application)
+
+    require "models/child_encrypted_pirate"
```

Comment:
This is needed to be done this way because we need a named class when we inheriting to properly set up filtered parameters (`"child_pirate.passphrase"`), but when using something like
```ruby
ChildPirate = Class.new(NamedPirate) do
  def self.name
    "ChildClass"
  end
end
```

`inherited` hook (the newly added `EncryptableRecord#inherited` hook specifically) will be called with an anonymous class and that won't work, see https://github.com/rails/rails/blob/12fc98fdbb1c5038c43aa8873d961cf5046505ed/activerecord/lib/active_record/encryption/configurable.rb#L53

Or there is a better way?

Reply:
Wouldn't something like this work?

```ruby
EncryptedPirate = Class.new(Pirate) do
   encrypts :catchphrase
end

# EncryptedPirate is a named class

ChildEncryptedPirate = Class.new(EncryptedPirate) # trigger the inherited callback
```

If we need to keep the `require`, maybe it's worth to add an assertion to confirm the constant hasn't be loaded at that point, to prevent other tests from requiring it causing an unwanted side effect: e.g: `assert_not defined?(ChildEncryptedAttribute), "Other test required this file?"`.

Reply:
👍 For assertion, would add that.

That is not working, because inherited is called with an unnamed class first as an argument, an *only after that* it is assigned to the `ChildEncryptedPirate` constant and thus receives a name.